hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -114,7 +115,17 @@ int QuectelNcpClient::init(const NcpClientConfig& conf) {
     conf_ = static_cast<const CellularNcpClientConfig&>(conf);
 
     // Initialize serial stream
-    auto sconf = SERIAL_8N1 | SERIAL_FLOW_CONTROL_NONE; // TODO: replace with SERIAL_FLOW_CONTROL_RTS_CTS when HW is fixed.
+    auto sconf = SERIAL_8N1 | SERIAL_FLOW_CONTROL_RTS_CTS;
+
+    // Hardware version
+    // V003 - 0x00 (disable hwfc)
+    // V004 - 0x01 (enable hwfc)
+    uint32_t hwVersion = 0xFF;
+    hal_get_device_hw_version(&hwVersion, nullptr);
+    if (hwVersion == 0x00) {
```

Comment:
Also, the revisions ideally should be declared somewhere, instead of using magic codes.

Reply:
What about adding this to `deviceid_hal.h`?
```
// Undefine hardware version
#define HAL_VERSION_UNDEFINED 0xFF
```

Reply:
That's fine, but I'd probably add B5 SoM hw revision v003 somewhere instead.

Reply:
Since it only belongs to B5SoM, I'll add `#define HAL_VERSION_B5SOM_V003 0x00` to `quectel_ncp_client.cpp`, do you think so?

Reply:
That's ok. Prefer constants vs defines though.

Reply:
Fixed